src/Sylius/Bundle/ShopBundle/Theme/ChannelBasedThemeContext.php

```diff
@@ -11,7 +11,7 @@
 
 declare(strict_types=1);
 
-namespace Sylius\Bundle\CoreBundle\Theme;
```

Comment:
This change is a BC break. this class should stay in `core` we should deprecate it and remove it in 2.0, also this class should be injected into the container conditionally based on `theme bundle` existence.

Reply:
It should be tested.

Reply:
And about the tests. There are tests for this class. I moved them to the shop bundle as well. Or what exactly should be tested?

Reply:
Given that service definition `sylius.theme.context.channel_based` didn't change, I would consider this a soft BC.

Class cannot be re-used directly, interface is in theme bundle, decoration will work without an issue, if service definition is overridden, then it will most likely use different class.

Though with 2.0 branch already existing, adding deprecation notice first might be a nice touch.

Reply:
if we place it before 2.0, then we should leave the class, deprecate it and make a duplication in ShopBundle. If going with 2.0 we can just move it. From the test perspective, we can just try to run tests/build container